pkg/resources/grant_application_role_acceptance_test.go

```diff
@@ -1,44 +1,58 @@
 package resources_test
 
 import (
-	"context"
 	"fmt"
-	"strings"
 	"testing"
 
 	acc "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance"
-	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
 	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
 
 	"github.com/hashicorp/terraform-plugin-testing/config"
-	"github.com/hashicorp/terraform-plugin-testing/helper/acctest"
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
-	"github.com/hashicorp/terraform-plugin-testing/terraform"
 	"github.com/hashicorp/terraform-plugin-testing/tfversion"
 )
 
+func createApp(t *testing.T, name string) *sdk.Application {
```

Comment:
this is a nice helper that should be reusable, please leave a comment to move it to helpers (we still need to figure out how we want to serve such complex helpers because know there are only single object methods there). Please create a subtask in SNOW-933289 with a brief description (and this discussion linked) and add the issue number to the TODO comment left here